Modix.Data/Repositories/InfractionRepository.cs

```diff
@@ -267,6 +277,33 @@ public async Task<bool> TryRescindAsync(long infractionId, ulong rescindedById)
             return true;
         }
 
+        /// <inheritdoc />
+        public async Task<bool> TryRestoreAsync(long infractionId, ulong restoredById)
+        {
+            var entity = await ModixContext.Infractions
+                .Where(x => x.Id == infractionId)
+                .FirstOrDefaultAsync();
+
+            if ((entity == null) || (entity.RescindActionId != null))
+            {
+                return false;
+            }
+
+            entity.RestoreAction = new ModerationActionEntity
+            {
+                GuildId = entity.GuildId,
+                Type = ModerationActionType.InfractionRestored,
+                Created = DateTimeOffset.Now,
+                CreatedById = restoredById,
+                InfractionId = entity.Id
+            };
+
+            await ModixContext.SaveChangesAsync();
+            await RaiseModerationActionCreatedAsync(entity.RescindAction);
```

Comment:
I think we need to modify the mod-log entry here, needs to defined in https://github.com/discord-csharp/MODiX/blob/9055b02859a2a8fceff6db0dddd2d134c2ffc408/Modix.Bot/Behaviors/ModerationLoggingBehavior.cs#L107-L126

Reply:
Added